### PR TITLE
Fix/visitor read record

### PR DESCRIPTION
**Pull request by Carolusian**

### CODE REVIEW

**Review by devformatters — APPROVED** [review]
Let's try it. 🪡

File: src/connectors/articleService.ts [hunk 1]
```diff
@@ -1290,7 +1290,10 @@ export class ArticleService extends BaseService {
     }
 
     // past record
-    const record = await this.baseFind({ where: { articleId, userId }, table })
+    // if visitor, distinguish base on ip address
+    const record = userId ?
+      await this.baseFind({ where: { articleId, userId }, table }) :
+      await this.baseFind({ where: { articleId, ip }, table })
```
Comment (guoliu) [inline src/connectors/articleService.ts:1296]: @Carolusian we are bundling all read records of visitors into a same record, and `ip` here is only recording the latest read. So it seems we shouldn't be finding records based on `ip`, but could do something like this:
```
  const record = userId ?
    await this.baseFind({ where: { articleId, userId }, table }) :
    await this.baseFind({ where: { articleId, userId: null }, table })
```
Which should be the same with the original code then.

**Review by guoliu — CHANGES_REQUESTED** [review]

File: src/connectors/articleService.ts (hunk 1 above)
Comment (Carolusian) [reply src/connectors/articleService.ts]: @guoliu 

```
  const record = userId ?
    await this.baseFind({ where: { articleId, userId }, table }) :
    await this.baseFind({ where: { articleId, userId: null }, table })
```
is equivalent to 
```
  const record = await this.baseFind({ where: { articleId, userId }, table })
```

The actual code cause the bug is:
```
if (readLength > MINUTE * 30) {
       return { newRead: false }
}
```

If we don't record visitor by IP, I updated the code to:
```
if (userId && readLength > MINUTE * 30) {
       return { newRead: false }
}
```

Comment (guoliu) [reply src/connectors/articleService.ts]: Ah I see, thank you!

**Comment by guoliu** [comment]
Looks good to me 🌳

**Review by guoliu — APPROVED** [review]

**Pull request merged**